crates/sui-types/src/base_types.rs

```diff
@@ -1099,25 +1111,60 @@ impl TxContext {
         self.epoch
     }
 
-    /// Derive a globally unique object ID by hashing self.digest | self.ids_created
-    pub fn fresh_id(&mut self) -> ObjectID {
-        let id = ObjectID::derive_id(self.digest(), self.ids_created);
+    pub fn sender(&self) -> SuiAddress {
+        SuiAddress::from(ObjectID(self.sender))
```

Comment:
I would recommend `self.sender.into()` or possibly `SuiAddress:from(self.sender)` here. Going through `ObjectID` to get there feels just wasteful lol